boa/src/builtins/array/mod.rs

```diff
@@ -1334,6 +1335,195 @@ impl Array {
         new_array.set_field("length", Value::from(new_array_len), true, context)?;
         Ok(new_array)
     }
+    /// `Array.prototype.splice ( start, [deleteCount[, ...items]] )`
+    ///
+    /// Splices an array by following
+    /// The deleteCount elements of the array starting at integer index start are replaced by the elements of items.
+    /// An Array object containing the deleted elements (if any) is returned.
+
+    pub(crate) fn splice(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        // 1. Let O be ? ToObject(this value).
+        let o = this.to_object(context)?;
+        // 2. Let len be ? LengthOfArrayLike(O).
+        let len = this.get_field("length", context)?.to_length(context)?;
```

Comment:
```suggestion
        let len = o.length_of_array_like(context)?;
```
we should call on `o` and use the already available `length_of_array_like`